saleor/checkout/complete_checkout.py

```diff
@@ -1053,21 +1053,33 @@ def _handle_allocations_of_order_lines(
     )
 
 
-def _handle_checkout_discount(order: "Order", checkout: "Checkout"):
+def _create_order_discount(order: "Order", checkout_info: "CheckoutInfo"):
+    checkout = checkout_info.checkout
     if checkout.discount:
-        # store voucher as a fixed value as it this the simplest solution for now.
-        # This will be solved when we refactor the voucher logic to use .discounts
-        # relations
-
-        order.discounts.create(
-            type=DiscountType.VOUCHER,
-            value_type=DiscountValueType.FIXED,
-            value=checkout.discount.amount,
-            name=checkout.discount_name,
-            translated_name=checkout.translated_discount_name,
-            currency=checkout.currency,
-            amount_value=checkout.discount_amount,
-        )
+        checkout_discount = checkout.discounts.first()
+        if checkout_discount and checkout_discount.type == DiscountType.ORDER_PROMOTION:
+            discount_data = model_to_dict(checkout_discount)
```

Comment:
From one type of object, you're creating different type of object. If `CheckoutDiscount` receives a new field, which will not be available on `OrderDiscount`, the `order.discount.create` will raise an exception, as you don't explicitly define which fields should be passed. 
If you don't want to explicitly provide each field, you can always define the fields that you want to receive from `model_to_dict` (`fields=`).
Then probably `del discount_data["checkout"]` will not be needed

Reply:
Because you explicitly say what fields you want to fetch. Defining the included fields will result in the same received data, no matter if the CheckoutDiscount object will be extended or not. Without this, all fields will be returned always.
Additionally, someone who checks the code, see what fields are exactly returned and needed.

Reply:
I get your point. However, I was thinking that way: if we extend `CheckoutDiscount` with some field, which doesn't fit to `OrderDiscount`, we will catch this error in tests. But if we extend `CheckoutDiscount` with a field, which fit to `OrderDiscount` we will need to update the list of fields, but no test will remind us about it. Does it make sense or not?

Reply:
> if we extend CheckoutDiscount with some field, which doesn't fit to OrderDiscount, we will catch this error in tests.
But in that case doesn't that mean, that you don't want to have this field in OrderDiscount? You have field A in CheckoutDiscount but not in OrderDiscount. 
If I am adding a new field only to CheckoutDiscount, the logic will be working as previously. If I expect that new field will be also propagated on OrderDiscount, I assume that I will add a new test or modify the existing one which will fail as the field is not propagated. 

My comment was only suggestions, so if you don't agree with me, we can also leave this as it is 😄